dpe/src/commands/sign.rs

```diff
@@ -118,4 +169,130 @@ mod tests {
+
+        // Bad handle.
+        assert_eq!(
+            Err(DpeErrorCode::InvalidHandle),
+            SignCmd {
+                handle: ContextHandle([0xff; ContextHandle::SIZE]),
+                label: TEST_LABEL,
+                flags: 0,
+                digest: TEST_DIGEST
+            }
+            .execute(&mut dpe, TEST_LOCALITIES[0])
+        );
+
+        // Wrong locality.
+        assert!(dpe
+            .get_active_context_pos(&ContextHandle::default(), TEST_LOCALITIES[0])
+            .is_some());
+        assert_eq!(
+            Err(DpeErrorCode::InvalidHandle),
+            SignCmd {
+                handle: ContextHandle::default(),
+                label: TEST_LABEL,
+                flags: 0,
+                digest: TEST_DIGEST
+            }
+            .execute(&mut dpe, TEST_LOCALITIES[1])
+        );
+
+        // Simulation contexts should not support the Sign command.
+        InitCtxCmd::new_simulation()
+            .execute(&mut dpe, TEST_LOCALITIES[0])
+            .unwrap();
+        assert!(dpe
+            .get_active_context_pos(&SIMULATION_HANDLE, TEST_LOCALITIES[0])
+            .is_some());
+        assert_eq!(
+            Err(DpeErrorCode::InvalidArgument),
+            SignCmd {
+                handle: SIMULATION_HANDLE,
+                label: TEST_LABEL,
+                flags: 0,
+                digest: TEST_DIGEST
+            }
+            .execute(&mut dpe, TEST_LOCALITIES[0])
+        );
+    }
+
+    #[test]
+    fn test_asymmetric_deterministic() {
+        let mut dpe =
+            DpeInstance::<OpensslCrypto>::new_for_test(SUPPORT, &TEST_LOCALITIES).unwrap();
+
+        for i in 0..3 {
+            DeriveChildCmd {
+                handle: ContextHandle::default(),
+                data: [i; DPE_PROFILE.get_hash_size()],
+                flags: DeriveChildCmd::MAKE_DEFAULT,
+                tci_type: i as u32,
+                target_locality: 0,
+            }
+            .execute(&mut dpe, TEST_LOCALITIES[0])
+            .unwrap();
+        }
+
+        let leaf_idx = dpe
+            .get_active_context_pos(&ContextHandle::default(), TEST_LOCALITIES[0])
+            .unwrap();
+
+        let cdi = dpe.derive_cdi(leaf_idx).unwrap();
+        let priv_key =
+            OpensslCrypto::derive_ecdsa_key(DPE_PROFILE.alg_len(), &cdi, &TEST_LABEL, b"ECC");
+
+        let cmd = SignCmd {
+            handle: ContextHandle::default(),
+            label: TEST_LABEL,
+            flags: 0,
+            digest: TEST_DIGEST,
+        };
+        let resp = match cmd.execute(&mut dpe, TEST_LOCALITIES[0]).unwrap() {
+            Response::Sign(resp) => resp,
+            _ => panic!("Incorrect response type"),
+        };
+
+        let sig = EcdsaSig::from_private_components(
+            BigNum::from_slice(&resp.sig_r_or_hmac).unwrap(),
+            BigNum::from_slice(&resp.sig_s).unwrap(),
+        )
+        .unwrap();
+        assert!(sig.verify(&TEST_DIGEST, &priv_key).unwrap());
```

Comment:
If it's not too much work, we should try to get the public key from CertifyKey and verify the signature with that to make sure the right public key is being output.

Could also do that test in the validation tests since it's more of a cross-command integration test.

Reply:
Done